src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java

```diff
@@ -65,22 +74,103 @@ public CCDCaseApi(
     }
 
     public Optional<Claim> getByExternalId(String externalId, String authorisation) {
-        final List<Claim> claims = search(authorisation, ImmutableMap.of("case.externalId", externalId));
+        return getCaseDetailsByExternalId(authorisation, externalId)
+            .map(CaseDetails::getData)
+            .map(this::convertToCCDCase)
+            .map((caseMapper::from));
+    }
 
-        if (claims.size() > 1) {
+    public Optional<Claim> linkDefendant(String externalId, String defendantId, String authorisation) {
+        User user = userService.authenticateAnonymousCaseWorker();
+        Optional<CaseDetails> optionalCaseDetails = getCaseDetailsByExternalId(authorisation, externalId);
```

Comment:
Will that give any result at all? I guess authorization header comes from the defendant which does not have access to case at this point yet.

Reply:
nope, was just fixing that